jupyter_rsession_proxy/__init__.py

```diff
@@ -45,7 +45,7 @@ def rewrite_auth(response, request):
 
 def setup_rserver():
     def _get_env(port):
-        return dict(USER=getpass.getuser())
+        return dict(USER=os.environ.get('NB_USER'))
```

Comment:
I think we should go through the following hierarchy:

1. If `JUPYTER_USER` is set, use that. That's the logged-in username, if using a JupyterHub
2. If `NB_USER` is set, use that. This defaults to `jovyan` I think?
3. Use `getpass.getuser()` if neither of these are set.

Reply:
Is `JUPYTER_USER` guaranteed to exist as a system user? I only see `JUPYTERHUB_USER` in PAWS, and that is not a system user in the container. 

The args to `--server-user` and `--user-identity` need to be a system user, as I understand it.

Reply:
Should the rsession-proxy use the above order, but have additional logic to verify that it is also a system user?

Reply:
sorry, I meant `JUPYTERHUB_USER` not `JUPYTER_USER`.

@vivian-rook I don't think it matters for rsession-proxy that it is a system user. Is that right, @ryanlovett?

Reply:
@yuvipanda I think the problem with #129 is that `Chicocvenancio` not a system user.

Reply:
ah goddamit, apparently i left my reading comprehension skills at home. Sorry!

In that case, can we try:

1. `pwd.getpwuid(os.getuid())[0]`, which should return a username that's in the `pwd` database. 
2. If this fails, then use `NB_USER`.

How does that sound?

Reply:
Something like this?